app/controllers/devise_token_auth/concerns/set_user_by_token.rb

```diff
@@ -94,15 +94,15 @@ def update_auth_header
     if @used_auth_by_token && !DeviseTokenAuth.change_headers_on_each_request
       # should not append auth header if @resource related token was
       # cleared by sign out in the meantime
-      return if @resource.reload.tokens[@token.client].nil?
```

Comment:
This line fails a spec that was added here https://github.com/lynndylanhurley/devise_token_auth/pull/531/files#diff-41be446a925979b6d8df8a6361c2ef48R343

Reply:
Maybe you can split this PR in 2, the second line can fix #1304 but this one needs a different approach, I'm thinking that we might need a good refactor with a different idea to avoid going to the database twice.

Reply:
Hi, 

Thank you for your responses. I fixed the spec. The problem is, I think they had made the second query intentionally. So, if your request has been processed but you log out in the meantime using the same access token then don't return the access_token back to any requests that contained the token. Makes sense... but, I don't think its that big of a deal. Because, the token is removed from the DB at this point and even if that DID happen, the clients shouldn't be able to access any of the system's resources using the access_token that was returned to them. Hope I make sense.

Reply:
Makes sense, I would recommend splitting this in 2 PRs, so #1304 is addressed and will be in the next version.